curriculum/challenges/english/25-front-end-development/workshop-fruit-search-app/67f90574263ebf363f9edd1d.md

```diff
@@ -0,0 +1,134 @@
+---
+id: 67f90574263ebf363f9edd1d
+title: Step 13
+challengeType: 0
+dashedName: step-13
+---
+
+# --description--
+
+Inside the `useEffect` hook, you should set a `700ms` delay before making a request to the Fruit API: `https://fruit-search.freecodecamp.rocks/api/fruits?q=${query}`. After fetching the data, map over the response and update the `results` state with an array of fruit names. Also, make sure to clear the timeout when the component re-renders or unmounts.
+
+# --hints--
+
+You should delay fetch from the fruit API with `setTimeout(700)`, map the fruit names to `setResults`, and clear the timeout.
+
+```js
+assert.match(
+  code,
+  /const\s+timeoutId\s*=\s*setTimeout\s*\(\s*async\s*\(\s*\)\s*=>\s*{[\s\S]*?fetch\(`https:\/\/fruit-search\.freecodecamp\.rocks\/api\/fruits\?q=\$\{query\}`\);[\s\S]*?setResults\(.*?\.map\(\s*\w+\s*=>\s*\w+\.name\s*\)\)[\s\S]*?},\s*700\s*\);[\s\S]*?return\s*\(\s*\)\s*=>\s*clearTimeout\(timeoutId\);/
+);
+```
+
+# --seed--
```

Comment:
The final solution (`--solutions--`) is missing at the end. Please look at the other workshops and lab so it can be added in. That is why the tests are failing